COMPILING.md

```diff
@@ -0,0 +1,308 @@
+   `-DCMAKE_BUILD_TYPE=Debug` (or `Release`) to the initial invocation.
+
+   If you *do* need to manually add flags, use `-DCMAKE_CXX_FLAGS=...` and
+   `-DCMAKE_EXE_LINKER_FLAGS=...`. This is useful for enabling clang's
+   sanitizers.
+
+   Finally, to enable building universal binaries on macOS, you can pass the
+   flag `-DCMAKE_OSX_ARCHITECTURES=i386;x86_64`. If you don't supply this flag,
+   the build will just be for the architecture of your machine.
+4. Run the build:
+   ```
+   cmake --build build
+   ```
+   This command tells CMake to invoke the correct tool to run the build in the
+   `build` directory. You can also use the build back-end directly by invoking
+   `make`, `ninja`, or opening the generated IDE project as appropriate.
+
+# WORKING WITH ECLIPSE
+
+To work with Eclipse, do the following:
+
+1. Select File -> New -> Makefile Project with Existing Code
+2. Type "cprover" as "Project Name"
+3. Select the "src" subdirectory as "Existing Code Location"
+4. Select a toolchain appropriate for your platform
+5. Click "Finish"
+6. Select Project -> Build All
+
+# CODE COVERAGE
+
+Code coverage metrics are provided using gcov and lcov. Ensure that you have
+installed lcov from http://ltp.sourceforge.net/coverage/lcov.php note for
+ubuntu lcov is available in the standard apt-get repos.
+
+To get coverage metrics run the following script from the regression directory:
+```
+get_coverage.sh
+```
+This will:
+ 1) Rebuild CBMC with gcov enabled
+ 2) Run all the regression tests
+ 3) Collate the coverage metrics
+ 4) Provide an HTML report of the current coverage
+
+# USING CLANG-FORMAT
+
+CBMC uses clang-format to ensure that the formatting of new patches is readable
+and consistent. There are two main ways of running clang-format: remotely, and
+locally.
+
+## RUNNING CLANG-FORMAT REMOTELY
+
+When patches are submitted to CBMC, they are automatically run through
+continuous integration (CI). One of the CI checks will run clang-format over
+the diff that your pull request introduces. If clang-format finds formatting
+issues at this point, the build will be failed, and a patch will be produced
+in the CI output that you can apply to your code so that it conforms to the
+style guidelines.
+
+To apply the patch, copy and paste it into a local file (`patch.txt`) and then
+run:
+```
+patch -p1 -i patch.txt
+```
+Now, you can commit and push the formatting fixes.
+
+## RUNNING CLANG-FORMAT LOCALLY
+
+### INSTALLATION
+
+To avoid waiting until you've made a PR to find formatting issues, you can
+install clang-format locally and run it against your code as you are working.
+
+Different versions of clang-format have slightly different behaviors. CBMC uses
+clang-format-3.8 as it is available the repositories for Ubuntu 16.04 and
+Homebrew.
+To install, run:
+```
+sudo apt install clang-format-3.8 # Run this on Ubuntu, Debian etc.
+brew install clang-format-3.8     # Run this on a Mac with Homebrew installed
+```
```

Comment:
We give build instructions for more platforms than just Ubuntu and Mac OS.  Any  chance of commands / pointers for those?

Reply:
I've updated with instructions for more platforms